From 2ddc418df57eae2ea98f60def28cfb113691bcd2 Mon Sep 17 00:00:00 2001
From: avaish1 <atul.vaish@intel.com>
Date: Wed, 15 Mar 2017 23:18:36 +0530
Subject: [PATCH 0401/1240] AndroidIA: Bluetooth Driver : Reducing Command
 Credit from Controller to 1 always

Intel BT Controller can not handle 2 Commands from the host at the same
time.Mostly they crashes and make BT unstable.
Hence reducing the command credit in all cases equal to 1.



Pairing/Connection/File Transfer/A2DP found to be working .
Signed-off-by: avaish1 <atul.vaish@intel.com>

Change-Id: I16fca67ea158e6ea9e0bef1572e7cd9ffc8c883b
---
 drivers/bluetooth/btusb.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)
 mode change 100644 => 100755 drivers/bluetooth/btusb.c

diff --git a/drivers/bluetooth/btusb.c b/drivers/bluetooth/btusb.c
old mode 100644
new mode 100755
index 86d7975a..7a16b8d
--- a/drivers/bluetooth/btusb.c
+++ b/drivers/bluetooth/btusb.c
@@ -1975,6 +1975,18 @@ static int btusb_recv_event_intel(struct hci_dev *hdev, struct sk_buff *skb)
 			}
 		}
 	}
+	else if (skb->len >= sizeof(struct hci_event_hdr)) {
+		struct hci_event_hdr *hdr;
+
+		hdr = (struct hci_event_hdr *) skb->data;
+
+		if (hdr->evt == HCI_EV_CMD_COMPLETE) {
+			*(__u8 *)(skb->data + 2) = 1;
+		} else if (hdr->evt == HCI_EV_CMD_STATUS) {
+			*(__u8 *)(skb->data + 3) = 1;
+		}
+    }
+
 
 	return hci_recv_frame(hdev, skb);
 }
-- 
2.7.4

